storage/store.go

```diff
@@ -1280,11 +1283,16 @@ func raftEntryFormatter(data []byte) string {
 // updateStoreStatus updates the store's status proto.
 func (s *Store) updateStoreStatus() {
 	now := s.ctx.Clock.Now().WallTime
-	s.status.UpdatedAt = now
-	s.status.RangeCount = int32(len(s.ranges))
+	status := &proto.StoreStatus{
+		StoreID:    s.Ident.StoreID,
+		NodeID:     s.Ident.NodeID,
+		UpdatedAt:  now,
+		RangeCount: int32(len(s.ranges)),
+		Stats:      proto.MVCCStats(s.scanner.Stats().MVCC),
+	}
+	atomic.StorePointer(&s.status, unsafe.Pointer(status))
 	key := engine.StoreStatusKey(int32(s.Ident.StoreID))
-	s.status.Stats = proto.MVCCStats(s.scanner.Stats().MVCC)
-	if err := s.ctx.DB.Run(client.PutProtoCall(key, s.status)); err != nil {
+	if err := s.ctx.DB.Run(client.PutProtoCall(key, status)); err != nil {
 		util.Error(err)
```

Comment:
What's this supposed to be doing doing? `util.Error` creates a new error. It doesn't do anything with it.